cpp/src/arrow/engine/substrait/util.h

```diff
@@ -0,0 +1,40 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include <memory>
+#include "arrow/engine/substrait/api.h"
+#include "arrow/util/iterator.h"
+#include "arrow/util/optional.h"
+
+namespace arrow {
+
+namespace substrait {
```

Comment:
```suggestion
namespace arrow {
namespace engine {
namespace substrait {
```

@vibhatha I think this may fix the Windows bug.  I think MSVC is getting confused between `::substrait::...` and `::arrow::substrait::...`.  Even if it doesn't this is a good change.  We are using `arrow::engine::substrait` everywhere else.  We should be using `arrow::engine::substrait` here too.

Reply:
Ah … I will update this and check.

Reply:
I added the modification. Waiting for the CIs...

Reply:
Looks like it resolved the issue.